composeApp/src/commonMain/kotlin/sp/bvantur/tasky/splash/presentation/SplashViewModel.kt

```diff
@@ -0,0 +1,9 @@
+package sp.bvantur.tasky.splash.presentation
+
+import androidx.lifecycle.ViewModel
+import sp.bvantur.tasky.splash.data.SplashRepository
+
+class SplashViewModel(private val splashRepository: SplashRepository) : ViewModel() {
+
+    fun isUserAuthorized(): Boolean = splashRepository.isUserAuthorized()
```

Comment:
Here you can see that to check if the user is authorised we had to click through the following classes:

1. `SplashViewModel` —> `SplashRepository`
2. `SplashRepository` —> `SplashLocalDataSource`
3. `SplashLocalDataSource` —> `SecurePersistentStorageProvider`

All of which just forwards to call to the next class

Reply:
💯